src/app/components/editors/gameobject/gameobject-template-addon/gameobject-template-addon.integration.spec.ts

```diff
@@ -0,0 +1,118 @@
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        GameobjectTemplateAddonModule,
+        RouterTestingModule,
+      ],
+    })
+      .compileComponents();
+  }));
+
+  function setup(creatingNew: boolean) {
+    handlerService = TestBed.get(GameobjectHandlerService);
+    handlerService['_selected'] = `${id}`;
+    handlerService.isNew = creatingNew;
+
+    queryService = TestBed.get(QueryService);
+    querySpy = spyOn(queryService, 'query').and.returnValue(of());
+
+    spyOn(queryService, 'selectAll').and.returnValue(of(
+      { results: creatingNew ? [] : [originalEntity] }
+    ));
+
+    fixture = TestBed.createComponent(GameobjectTemplateAddonComponent);
+    component = fixture.componentInstance;
+    page = new GameobjectTemplateAddonPage(fixture);
+    fixture.autoDetectChanges(true);
+    fixture.detectChanges();
+  }
+
+  describe('Creating new', () => {
+    beforeEach(() => setup(true));
+
+    it('should correctly initialise', () => {
+      page.expectQuerySwitchToBeHidden();
+      page.expectFullQueryToBeShown();
+      page.expectFullQueryToContain(expectedFullCreateQuery);
+    });
+
+    it('changing a property and executing the query should correctly work', () => {
+      const expectedQuery = 'DELETE FROM `gameobject_template_addon` WHERE (`entry` = ' + id + ');\n' +
+      'INSERT INTO `gameobject_template_addon` (`entry`, `faction`, `flags`, `mingold`, `maxgold`) VALUES\n' +
+      '(' + id + ', 35, 0, 0, 0);';
+
+      querySpy.calls.reset();
+
+      page.setInputValueById('faction', '35');
+      page.clickExecuteQuery();
+
+      page.expectFullQueryToContain(expectedQuery);
+      expect(querySpy).toHaveBeenCalledTimes(1);
+      expect(querySpy.calls.mostRecent().args[0]).toContain(expectedQuery);
+    });
+  });
+
+  describe('Editing existing', () => {
+    beforeEach(() => setup(false));
+
+    it('should correctly initialise', () => {
+      page.expectDiffQueryToBeShown();
+      page.expectDiffQueryToBeEmpty();
+      page.expectFullQueryToContain(expectedFullCreateQuery);
+    });
+
+    it('changing all properties and executing the query should correctly work', () => {
+      const expectedQuery = 'UPDATE `gameobject_template_addon` SET `flags` = 1, `mingold` = 2, `maxgold` = 3 WHERE (`entry` = 1234);';
+
+      querySpy.calls.reset();
+
+      page.changeAllFields(originalEntity, ['VerifiedBuild']);
+      page.clickExecuteQuery();
+
+      page.expectDiffQueryToContain(expectedQuery);
+      expect(querySpy).toHaveBeenCalledTimes(1);
+      expect(querySpy.calls.mostRecent().args[0]).toContain(expectedQuery);
+    });
+
+    it('changing values should correctly update the queries', () => {
+      page.setInputValueById('faction', '35');
+      page.expectDiffQueryToContain(
+        'UPDATE `gameobject_template_addon` SET `faction` = 35 WHERE (`entry` = ' + id + ');'
+      );
+      page.expectFullQueryToContain('35');
+
+    });
```

Comment:
add one test case that uses the FlagsSelector to change the value, you can look at this example:

https://github.com/azerothcore/Keira3/blob/master/src/app/components/editors/quest/quest-template-addon/quest-template-addon.integration.spec.ts#L165